adapters/shp.py

```diff
@@ -72,6 +72,10 @@ def read(fp, prop_map, source_filename=None):
 
         for rec in source:
             transformed = _transformer(source.crs, rec)
+            if filterer is not None and not filterer.keep(transformed):
```

Comment:
filter is a built in function in python, so naming a variable that is problematic. Hence filterer